lib/Twig/Extensions/TokenParser/Trans.php

```diff
@@ -25,33 +25,40 @@ public function parse(Twig_Token $token)
         $plural = null;
         $notes = null;
 
+        $withVars = new \Twig_Node_Expression_Array(array(), $lineno);
```

Comment:
this should be ``null`` if there is no ``with`` token, not an empty array, to avoid doing a useless merging during the compilation